packages/base/card-api.gts

```diff
@@ -2904,3 +2904,120 @@ type ElementType<T> = T extends (infer V)[] ? V : never;
 function makeRelativeURL(maybeURL: string, opts?: SerializeOpts): string {
   return opts?.maybeRelativeURL ? opts.maybeRelativeURL(maybeURL) : maybeURL;
 }
+
+type EmptySchema = {};
+
+type ArraySchema = {
+  type: 'array';
+  items: Schema;
+};
+
+type ObjectSchema = {
+  type: 'object';
+  properties: {
+    [fieldName: string]: Schema;
+  };
+};
+
+type DateSchema = {
+  type: 'string';
+  format: 'date' | 'date-time';
+};
+
+type PrimitiveSchema = {
+  type: 'string' | 'number' | 'boolean';
+};
+
+type Schema =
+  | EmptySchema
+  | ArraySchema
+  | ObjectSchema
+  | DateSchema
+  | PrimitiveSchema;
+
+/**
+ * Generates the JSON schema required for patching
+ *
+ * @param def - The BaseDef to generate the patch call specification for.
+ * @param typeChain - A list of the types
+ */
+function _generatePatchCallSpecification(
+  def: typeof BaseDef,
+  typeChain: string[],
+) {
+  // If we have already seen this type, then we are in a recursive loop
+  // and we should stop. This can happen when a card has a field that
+  // references itself.
+  if (typeChain.includes(def.name)) {
+    return undefined;
+  } else {
+    // This must be a copy, otherwise we're mutating an array
+    // used through the recursive calls
+    typeChain = [...typeChain, def.name];
+  }
+  // An explicit list of types that we will support in the patch call
+  if (primitive in def) {
+    switch (def.name) {
+      case 'NumberField':
+        return { type: 'number' };
+      case 'StringField':
+        return { type: 'string' };
+      case 'BooleanField':
+        return { type: 'boolean' };
+      case 'DateField':
+        return { type: 'string', format: 'date' };
+      case 'DateTimeField':
+        return { type: 'string', format: 'date-time' };
+      default:
+        return undefined;
```

Comment:
there are other types of primitives like the `CodeRef` field. ultimately you will not be able to make an exhaustive list of all the primitive fields since primitives can be user provided code